src/core/imageUtils.js

```diff
@@ -0,0 +1,9 @@
+import { isAllowedOrigin } from 'core/utils';
+import fallbackIcon from 'amo/img/icons/default-64.png';
+
+export function getAddonIconUrl(addon) {
+  if (!addon) {
+    throw new Error('addon cannot be empty');
```

Comment:
Ah, you were right. This should not throw an error. In the case of a falsy addon, the `getAddonIconUrl()` function should return the `fallbackIcon`